### PR TITLE
CMake: Fail if there is no analytical parser

### PR DESCRIPTION
**Pull request by mwestphal**
Fix https://github.com/openturns/openturns/issues/2839

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -219,6 +219,9 @@ if (USE_EXPRTK)
   set (OPENTURNS_HAVE_ANALYTICAL_PARSER TRUE)
   set (SYMBOLICPARSER_DEFAULT_BACKEND ExprTk)
 endif ()
+if (NOT OPENTURNS_HAVE_ANALYTICAL_PARSER)
+  message(FATAL_ERROR "Analytical paser required, please enable either USE_MUPARSER or USE_EXPRTK")
```
Comment (mwestphal) [inline CMakeLists.txt:223]:
```suggestion
  message(FATAL_ERROR "Analytical parser required, please enable either USE_MUPARSER or USE_EXPRTK")
```

**Review by jschueller — APPROVED** [review]

**Review by regislebrun — APPROVED** [review]

**Comment by regislebrun** [comment]
@jschueller I was pretty sure that ExprTk was always available (as we include its unique source file in our source tree) and that MuParser was an additional capability. As I am not very fluent in CMake, could you please tell me if it is possible to have both parsers or if we have to choose at compile time which parser will be used?

**Comment by mwestphal** [comment]
> @jschueller I was pretty sure that ExprTk was always available (as we include its unique source file in our source tree) and that MuParser was an additional capability. As I am not very fluent in CMake, could you please tell me if it is possible to have both parsers or if we have to choose at compile time which parser will be used?

That's definitely possible to implement such logic. I could use muparser when `USE_MUPARSER=ON` and fallback to exprtk when `USE_MUPARSER=OFF`.

**Comment by regislebrun** [comment]
> > @jschueller I was pretty sure that ExprTk was always available (as we include its unique source file in our source tree) and that MuParser was an additional capability. As I am not very fluent in CMake, could you please tell me if it is possible to have both parsers or if we have to choose at compile time which parser will be used?
> 
> That's definitely possible to implement such logic. I could use muparser when `USE_MUPARSER=ON` and fallback to exprtk when `USE_MUPARSER=OFF`.

It would be a perfect solution. BTW I assumed that at least one parser was available in several places in the code, e.g in PosteriorDistribution when I compute the moments by integration. It would be a pity to only have a conditional access to these computations despite the inclusion of ExprTk source code!

**Pull request merged**